core/node/provider.go

```diff
@@ -5,145 +5,151 @@ import (
+			provider.KeyProvider(keyProvider),
+			provider.ThroughputReport(func(reprovide bool, complete bool, keysProvided uint, duration time.Duration) bool {
+				avgProvideSpeed := duration / time.Duration(keysProvided)
+				count := uint64(keysProvided)
+
+				if !reprovide || !complete {
+					// We don't know how many CIDs we have to provide, try to fetch it from the blockstore.
+					// But don't try for too long as this might be very expensive if you have a huge datastore.
+					ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+					defer cancel()
+
+					// FIXME: I want a running counter of blocks so size of blockstore can be an O(1) lookup.
+					ch, err := bs.AllKeysChan(ctx)
+					if err != nil {
+						logger.Errorf("fetching AllKeysChain in provider ThroughputReport: %v", err)
+						return false
+					}
+					count = 0
+				countLoop:
+					for {
+						select {
+						case _, ok := <-ch:
+							if !ok {
+								break countLoop
+							}
+							count++
+						case <-ctx.Done():
+							// really big blockstore mode
+
+							// how many blocks would be in a 10TiB blockstore with 128KiB blocks.
+							const probableBigBlockstore = (10 * 1024 * 1024 * 1024 * 1024) / (128 * 1024)
+							// How long per block that lasts us.
+							expectedProvideSpeed := reprovideInterval / probableBigBlockstore
+							if avgProvideSpeed > expectedProvideSpeed {
+								logger.Errorf(`
+🔔🔔🔔 YOU MAY BE FALLING BEHIND DHT REPROVIDES! 🔔🔔🔔
+
+⚠️ Your system might be struggling to keep up with DHT reprovides!
+Meaning your content being partially or completely inacessible on the network.
+We observed that you recently provided %d keys at an average rate of %v per key.
+
+🕑 An attempt to estimate your blockstore size timed out after 5 minutes,
+implying your blockstore might be exceedingly large. Assuming a considerable
+size of 10TiB, it would take %v to provide the complete set.
+
+⏰ The total provide time needs to stay under 24 hours to prevent falling behind!
+
+💡 Consider enabling the Accelerated DHT to enhance your system performance. See:
+https://github.com/ipfs/kubo/blob/master/docs/config.md#routingaccelerateddhtclient`,
+									keysProvided, avgProvideSpeed, avgProvideSpeed*probableBigBlockstore)
+								return false
+							}
+						}
+					}
+				}
+
+				// How long per block that lasts us.
+				expectedProvideSpeed := reprovideInterval / time.Duration(count)
+				if avgProvideSpeed > expectedProvideSpeed {
+					// FIXME(@Jorropo): add link to the accelerated DHT client docs once this isn't experimental anymore.
+					logger.Errorf(`
+🔔🔔🔔 YOU ARE FALLING BEHIND DHT REPROVIDES! 🔔🔔🔔
+
+⚠️ Your system is struggling to keep up with DHT reprovides!
+Meaning your content being partially or completely inacessible on the network.
+We observed that you recently provided %d keys at an average rate of %v per key.
+
+💾 Your total CID count is ~%d which would total at %v reprovide process.
```

Comment:
```suggestion
💾 Your total CID count is ~%d which would total at %v UNITS to reprovide.
```

UNITS should become whatever unit of measurement we're working with here.  Seconds, millis, minutes?  Lets be clear.

Reply:
`%v` will call `.String` on the type if it implements `fmt.Stringer`, `time.Duration` implements it by serialising with the unit https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/time/time.go;l=644.

Example:
```go
fmt.Printf("%v", time.Second * 27 + time.Minute) // prints: 1m27s
```